### PR TITLE
Add <details> and <summary> elements

### PR DESCRIPTION
**Pull request by kimulaco**
I have added a disclosure widget with `<details>` and `<summary>` to the demo page.
Although the display on IE and Edge has not been confirmed yet, I would like to first confirm the design.
Please let me know if there are any points you would like to change.

In the future, it will be displayed in an open state in the unsupported browser.

Edit: I forgot to associate the issue. https://github.com/kognise/water.css/issues/30

### CODE REVIEW

**Comment by jonaskuske** [comment]
> Although the display on IE and Edge has not been confirmed yet

I assume it'll look quite awkward as they don't have support for it?  
But not really our concern imo, we're just styling the elements the browser knows, it's up to the developers to only use HTML that is valid in their targeted browsers. (or to load respective polyfills)

**Comment by jonaskuske** [comment]
But I think it looks good! 👍🏻

**Comment by kimulaco** [comment]
It is not supported by IE or Edge and will not work.
So I think there is no need for full support.

However, I think that at least you need to be able to view the content, so I will do additional checks.

**Comment by jonaskuske** [comment]
That's true!

File: src/parts/_misc.css [hunk 1]
```diff
@@ -38,7 +38,7 @@ tbody tr:nth-child(even) {
   border-radius: 6px;
 }
 
-::-webkit-scrollbar-thumb {
+::-webkit-scrollbar-thumb {
```
Comment (kylejrp) [inline src/parts/_misc.css]: There's a weird character between the `b` and `a` in `-scrollbar`:
image

**Comment by kimulaco** [comment]
@kylejrp Oh... Sorry.
I was not careful enough...!
Since I corrected, can you confirm?

**Comment by jonaskuske** [comment]
Adding `display: flex; flex-direction: column;` to the `<details>` element fixes the styling in browsers like IE and Edge because the default alignment of flex-children is `stretch` (the version on the bottom is the fixed one):

image

We could also consider only stretching the summary to the full width of the `<details>` container, so all other children will keep their intrinsic width, like:
```css
details {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}
summary {
  align-self: stretch;
}
```

**Comment by jonaskuske** [comment]
A better (and spec-compliant) alternative:

```css
summary {
  display: list-item;
}
```

This will make the summary stretch to fill the full width of the `<details>` element.
Summary having a display mode of `list-item` [is part of the spec anyhow](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary#Default_style), but not all browsers have the correct UA styles. (and setting it to anything other than `list-item`, like `block`, also stretches it to full width but breaks the display in Firefox)

**Comment by kylejrp** [comment]
My apologies @kimulaco, I missed that we had a pending merge request for this and and merged in #106. 

Are you able to compare your changes here to #106 and make a new pull request for any improvements that could be made?

I really value the work you've done here and I'd love to see your changes included in water.css. I'm gong to close this merge request for now since we now have an implementation in master but would love to see you continue to contribute. Again, my sincere apologies for my mistake.

**Pull request closed without merging**